### ISSUE TITLE
Search bars everywhere

### ISSUE THREAD

**Issue by OxygenCobalt** [OxygenCobalt/Auxio#288]
Thinking about re-working the search bar in the app to available "everywhere". This way, you could search an album, artist, genre, or playlist, and not just the whole library. The only exception would be the queue, in which I really do not believe search makes sense.

This would require some UI reworks and more crazy magic to get it working in a fluid and sensible way. Most notably, the toolbar in the app would be replaced with a persistent SearchBar like this:

![image](https://user-images.githubusercontent.com/65370175/205367087-bf419432-3aa8-4392-b6f6-9508949c986d.png)

As you navigate around, the hint text would change to align with the current item you are viewing. For example, "Search your music library" would turn into "Search this artist"/"Search [Artist Name]" when you navigate into an artist.

No guarantees I implement this. Please let me know in the comments if you like this idea.

**Comment by illdeletethis** [comment]
would not get any benefit or disadvantage from this

when going into a category i know why i"m there, search helps finding something i"m in some way uncertain of.

in this sense, the queue is the only place i might get a benefit from, for quickly finding a position to jump to, in the one place without an immediately apparent order, but not enough to feel strongly about it.

should others consider it a worthwhile feature i have nothing against it though

**Comment by OxygenCobalt** [comment]
Thanks for the feedback. I'm considering this because I know that some users might want to search through playlists (#36) when they are added. This is all speculation however.

**Comment by OxygenCobalt** [comment]
> there is one big disadvantage i can see, lets say you were looking for an item in a different category than you meant to, now you have to delete your query, exit search and type it in again in the proper search bar. i think it would be maybe better if the search results had a chip bar "all" "songs" "albums" "artists" ...

You're right about this. I know that the "omnibar" in Google Messages can be accessed at multiple points, yet it still re-directs to the same search view with some chip value applied to denote what you are searching. I'll try to do this.

> but i understand if this adds too much complexity, so no search in queue is fine.

It's not *complicated*, but it is hard to integrate. Placing it anywhere but the actual queue view feels really disjoint, but there's no *perceived* room left in the queue to place it in without it feeling cluttered. Plus, I tend to not jump very far ahead in the queue to play a specific song, so such functionality is not priority for me.

**Comment by OxygenCobalt** [comment]
There is one caveat with what I described prior: I think it would require me to change the "click" action to no longer go to the album/artist/genre, but instead sub-filter the results. I'm not sure how desirable this is. Perhaps I'll add a menu option.

**Comment by OxygenCobalt** [comment]
Another perk here is that I could hypothetically make music loading state global.
